ape_safe/accounts.py

```diff
@@ -208,7 +208,63 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
         # NOTE: Need to override `AccountAPI` behavior for balance checks
         return self.provider.prepare_transaction(txn)
 
-    def call(self, txn: TransactionAPI, **call_kwargs) -> ReceiptAPI:  # type: ignore[override]
+    def _impersonated_call(self, txn: TransactionAPI, **safe_tx_kwargs) -> ReceiptAPI:
+        safe_tx = self.create_safe_tx(txn, **safe_tx_kwargs)
+        safe_tx_exec_args = list(safe_tx._body_["message"].values())
+        signatures = []
+        # Bypass signature collection logic and attempt to submit by impersonation
+        # NOTE: Only works for fork and local networks
+        if self.confirmations_required > 1:
+            safe_tx_hash = self.contract.getTransactionHash(*safe_tx_exec_args)
+            for signer_address in self.signers[: self.confirmations_required - 1]:
+                impersonated_signer = self.account_manager.test_accounts[signer_address]
+                self.contract.approveHash(safe_tx_hash, sender=impersonated_signer)
+                signatures.append(
+                    MessageSignature(
+                        v=1,  # Approved hash (e.g. submitter is approved)
+                        r=b"\x00" * 12 + to_bytes(hexstr=impersonated_signer.address),
+                        s=b"\x00" * 32,
+                    )
+                )
+
+        impersonated_sender = self.account_manager.test_accounts[
+            self.signers[self.confirmations_required - 1]
+        ]
+        signatures.append(
+            MessageSignature(
+                v=1,  # Approved hash (e.g. submitter is approved)
+                r=b"\x00" * 12 + to_bytes(hexstr=impersonated_sender.address),
+                s=b"\x00" * 32,
+            )
+        )
+
+        # NOTE: Must order signatures in ascending order of signer address (converted to int)
+        def sigr2int(sig: MessageSignature) -> int:
+            return to_int(sig.r)
+
+        encoded_signatures = b"".join(sig.encode_rsv() for sig in sorted(signatures, key=sigr2int))
```

Comment:
would it make sense to do an earlier pre-sort of the owners used?

Reply:
it doesn't help or hurt any, the signatures have to be ordered, but the approvals can be peformed in any order

Reply:
the property returning sorted owners (`.signers`) might be even better, safe [requires](https://github.com/safe-global/safe-contracts/blob/e870f514ad34cd9654c72174d6d4a839e3c6639f/contracts/Safe.sol#L331) them in sorted owners anyway

Reply:
`.local_signers` won't be sorted because will add a config that allows re-ordering them according to some user-desired ordering